planner/core/logical_plan_builder.go

```diff
@@ -2516,12 +2516,13 @@ func extractTableSourceAsNames(node ast.ResultSetNode, input []string, onlySelec
 	return input
 }
 
-func appendVisitInfo(vi []visitInfo, priv mysql.PrivilegeType, db, tbl, col string) []visitInfo {
+func appendVisitInfo(vi []visitInfo, priv mysql.PrivilegeType, db, tbl, col string, err error) []visitInfo {
```

Comment:
It seems that the value passed to the parameter of `err` is `ErrPrivilegeCheckFail`. If that is true, could we remove this argument and set the `visitInfo.err` to `ErrPrivilegeCheckFail` in line2525?

Reply:
It is only `ErrPrivilegeCheckFail` for this refactor.  It will be different error codes in subsequent pushes.

In fact, using `ErrPrivilegeCheckFail` should not be used, since it is not a MySQL compatible error.